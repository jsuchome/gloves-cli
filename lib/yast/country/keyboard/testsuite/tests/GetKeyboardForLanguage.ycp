/**
 * tests for Keyboard::GetKeyboardForLanguage
 * Author: jsuchome@suse.cz
 * $Id: GetKeyboardForLanguage.ycp 27936 2006-02-13 20:01:14Z olh $
 */
{

    // testedfiles: Keyboard.ycp

    include "testsuite.ycp";

    map READ = $[
	"sysconfig" : $[
	    "keyboard" : $[
		"YAST_KEYBOARD"	: "czech-qwerty,pc104",
	    ],
	],
	"target" : $[
	    "size"		: 1,
	    "yast2"		: $[],
	    "tmpdir"		: "/tmp",
	    "ycp"		: $[],
	],
	"probe"	: $[
	    "architecture"	: "i386"
	],
    ];
    map EXECUTE	= $[
	"target" : $[
	    "bash_output"	: $[],
	],
    ];

    TESTSUITE_INIT([READ,$[],EXECUTE], nil);

    import "Keyboard";

    map R = $[
	"target" : $[
	    "yast2"		: $[
		"en_US"		: "english-us",
		"cs_CZ"		: "czech",
		"de_CH"		: "german-ch",
	    ],
	],
    ];

    TEST (``(Keyboard::GetKeyboardForLanguage ("en_US", "en_US")),
	[R,$[],$[]], []);

    TEST (``(Keyboard::GetKeyboardForLanguage ("cs_CZ", "en_US")),
	[R,$[],$[]], []);

    TEST (``(Keyboard::GetKeyboardForLanguage ("de_CH", "en_US")),
	[R,$[],$[]], []);

    TEST (``(Keyboard::GetKeyboardForLanguage ("de_DE", "en_US")),
	[R,$[],$[]], []);
}
