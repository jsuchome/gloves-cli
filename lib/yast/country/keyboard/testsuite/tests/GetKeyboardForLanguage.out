Read	.target.yast2 "keyboard_raw.ycp" $[]
Read	.target.yast2 "lang2keyboard.ycp" $["cs_CZ":"czech", "de_CH":"german-ch", "en_US":"english-us"]
Return	english-us
Read	.target.yast2 "lang2keyboard.ycp" $["cs_CZ":"czech", "de_CH":"german-ch", "en_US":"english-us"]
Return	czech
Read	.target.yast2 "lang2keyboard.ycp" $["cs_CZ":"czech", "de_CH":"german-ch", "en_US":"english-us"]
Return	german-ch
Read	.target.yast2 "lang2keyboard.ycp" $["cs_CZ":"czech", "de_CH":"german-ch", "en_US":"english-us"]
Return	en_US
